device/protocol_mqtt_v1.go

```diff
@@ -53,7 +53,6 @@ func (d *Device) initializeMQTTClient() error {
 			if d.OnErrors != nil {
 				d.OnErrors(d, errors.New(errorMsg))
 			}
-			d.m.Disconnect(0)
 			fmt.Println(errorMsg)
 			return
```

Comment:
Maybe add a comment above `return`, something like `// If we failed to execute this action it means that we got disconnected, the reconnection mechanism will take care of that so we just return`